File: projects/developer-terminal-c-and-vb/Developer Term Csharp Beta r02/yteShooter.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace DeveloperTerminal
{
    public partial class ByteShooter : Form
    {
        #region initialization
        private int Interval = 100;

        public ByteShooter()
        {
            InitializeComponent();
        }

        private void ByteShooter_Load(object sender, EventArgs e)
        {
            char c;
            // Load the comboBox with 0x00 to 0xFF
            for (int i = 0; i < 16; i++)
            {
                c = (char)i;
                this.comboBoxSendByte.Items.Add("0x0" + i.ToString("X") + " -> " +  c.ToString());
            }
            for (int i = 16; i < 256; i++)
            {
                c = (char)i;
                this.comboBoxSendByte.Items.Add("0x" + i.ToString("X") + " -> " + c.ToString());
            }
        }
        #endregion

        #region Data Assessors
        public int interval
        {
            get
            {
                return Interval;
            }
        }

        private byte ByteToSend = 0;
        public byte byteToSend
        {
            get
            {
                return ByteToSend;
            }
        }
        #endregion

        #region Components
        private void textBox1_TextChanged(object sender, EventArgs e)
        {
            try
            {
                Interval = Convert.ToInt32(textBox1.Text);
                timer1.Interval = Interval;
            }
            catch
            {
                timer1.Enabled = false;
                MessageBox.Show("Error with interval");
            }
        }

        private void comboBoxSendByte_SelectedIndexChanged(object sender, EventArgs e)
        {      
            ByteToSend = (byte)comboBoxSendByte.SelectedIndex;
        }

        private void buttonSend_Click(object sender, EventArgs e)
        {
            this.DialogResult = DialogResult.OK;
        }

        private void buttonCancel_Click(object sender, EventArgs e)
        {
            this.DialogResult = DialogResult.Cancel;
        }
        #endregion

    }
}

File: projects/developer-terminal-c-and-vb/Developer Term VB Beta r02/yteShooter.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Namespace DeveloperTerminal
	Public Partial Class ByteShooter
		Inherits Form
		#Region "initialization"
		Private Interval_Renamed As Integer = 100

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub ByteShooter_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Dim c As Char
			' Load the comboBox with 0x00 to 0xFF
			For i As Integer = 0 To 15
				c = ChrW(i)
				Me.comboBoxSendByte.Items.Add("0x0" & i.ToString("X") & " -> " & c.ToString())
			Next i
			For i As Integer = 16 To 255
				c = ChrW(i)
				Me.comboBoxSendByte.Items.Add("0x" & i.ToString("X") & " -> " & c.ToString())
			Next i
		End Sub
		#End Region

		#Region "Data Assessors"
		Public ReadOnly Property interval() As Integer
			Get
				Return Interval_Renamed
			End Get
		End Property

		Private ByteToSend_Renamed As Byte = 0
		Public ReadOnly Property byteToSend() As Byte
			Get
				Return ByteToSend_Renamed
			End Get
		End Property
		#End Region

		#Region "Components"
		Private Sub textBox1_TextChanged(ByVal sender As Object, ByVal e As EventArgs) Handles textBox1.TextChanged
			Try
				Interval_Renamed = Convert.ToInt32(textBox1.Text)
				timer1.Interval = Interval_Renamed
			Catch
				timer1.Enabled = False
				MessageBox.Show("Error with interval")
			End Try
		End Sub

		Private Sub comboBoxSendByte_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles comboBoxSendByte.SelectedIndexChanged
			ByteToSend_Renamed = CByte(comboBoxSendByte.SelectedIndex)
		End Sub

		Private Sub buttonSend_Click(ByVal sender As Object, ByVal e As EventArgs) Handles buttonSend.Click
			Me.DialogResult = System.Windows.Forms.DialogResult.OK
		End Sub

		Private Sub buttonCancel_Click(ByVal sender As Object, ByVal e As EventArgs) Handles buttonCancel.Click
            Me.DialogResult = DialogResult.Cancel
		End Sub
		#End Region

	End Class
End Namespace